### PR TITLE
Additions and changes to the Plugin object and its reference global.

### PR DESCRIPTION
**Pull request by matkeg**
## Changes

Added additional information to the Plugin object and its reference global.  

(Note for the reviewer: Please check if the ```<Alert>``` tag and the code snippets can be used just like I used them, if they can't be used like this, please modify the content.)

https://devforum.roblox.com/t/plugin-global-doesnt-exist-in-module-script-under-plugin-script/659007?u=matkeftw

## Checks

By submitting your pull request for review, you agree to the following:

- [X] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [X] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [X] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

**Review by metatablecat — COMMENTED** [review]
Plugins are hosted in a hirearchical manner, where the source code is placed directly under it.

```
Plugin
  your_code.rbxm
    Script
    AModule
```

It might be easier, for the sake of these docs (though probably slower), to use a method like FindFirstAncestorOfClass("Plugin")

**Comment by IgnisRBX** [comment]
Hi @matkeg , apologies for the delay. I'm looking at this PR this week and I'll leave a couple comments soon.

File: content/en-us/reference/engine/classes/Plugin.yaml [hunk 1]
```diff
@@ -4,9 +4,58 @@ category: Plugins
...
+  end
+  
+  return pluginModule
+  ```
+
+  ```lua title='Output'
+    -- Plugin reference is set!
+  ```
+
 code_samples:
```
Comment (IgnisRBX) [inline content/en-us/reference/engine/classes/Plugin.yaml:59]:
```suggestion
  **Plugin** is the main object responsible for creating basic Studio widgets,
  plugin toolbars, plugin buttons, and more. It is a custom add-on to Studio
  which adds new behavior and features that are not normally included. The
  **Plugin** object can be accessed through the `Global.RobloxGlobals.plugin`
  global reference in a `Class.Script` that is executed as a plugin.
code_samples:
  - Plugin-Global-Pass
  - Plugin-Global-Receive
```

Comment (IgnisRBX) [reply content/en-us/reference/engine/classes/Plugin.yaml:59]: In a separate PR, I made your code samples into linkable files and put them in the `code_samples` section.

File: content/en-us/reference/engine/globals/RobloxGlobals.yaml [hunk 2]
```diff
@@ -33,9 +33,16 @@ properties:
...
+      A reference to the `Class.Plugin` object that represents the plugin being run
+      from this `Class.Script`. This reference exists only in the context where
+      a script is executed as a plugin and is not passed to `Class.ModuleScript`s
+      within the plugin. To use this reference in a `Class.ModuleScript`, you must
+      explicitly pass it.
+      
+      ```lua title='Check if the script is run as a plugin.'
+      assert(plugin, "This script must be run as a plugin.")
+      -- Code beyond this point will execute only if the script is run as a plugin.
+      ```
```
Comment (IgnisRBX) [inline content/en-us/reference/engine/globals/RobloxGlobals.yaml:45]:
```suggestion
      A reference to the `Class.Plugin` object that represents the plugin being run
      from this `Class.Script`. This reference exists only in the context where
      a script is executed as a plugin and is not passed to
      `Class.ModuleScript|ModuleScripts` within the plugin. To use this reference
      in a `Class.ModuleScript`, you must explicitly pass it.
      
      ```lua
      assert(plugin, "This script must be run as a plugin!")
      -- Code beyond this point will execute only if the script is run as a plugin
      ```
```

**Review by IgnisRBX — APPROVED** [review]

**Pull request merged**